optuna_dashboard/artifact/_backend.py

```diff
@@ -178,32 +190,38 @@ def objective(trial: optuna.Trial) -> float:
         "encoding": encoding or guess_encoding,
         "filename": filename,
     }
-    attr_key = _artifact_prefix(trial_id=trial_id) + artifact_id
-    storage.set_study_system_attr(study_id, attr_key, json.dumps(artifact))
+    attr_key = ARTIFACTS_ATTR_PREFIX + artifact_id
+    storage.set_trial_system_attr(trial_id, attr_key, json.dumps(artifact))
 
     with open(file_path, "rb") as f:
         backend.write(artifact_id, f)
     return artifact_id
 
 
 def _artifact_prefix(trial_id: int) -> str:
-    return ARTIFACTS_ATTR_PREFIX + f"{trial_id}:"
+    return DASHBOARD_ARTIFACTS_ATTR_PREFIX + f"{trial_id}:"
 
 
 def get_artifact_meta(
     storage: BaseStorage, study_id: int, trial_id: int, artifact_id: str
 ) -> Optional[ArtifactMeta]:
-    study_system_attr = storage.get_study_system_attrs(study_id)
+    # Search study_system_attrs due to backward compatibility.
+    study_system_attrs = storage.get_study_system_attrs(study_id)
     attr_key = _artifact_prefix(trial_id=trial_id) + artifact_id
-    artifact_meta = study_system_attr.get(attr_key)
+    artifact_meta = study_system_attrs.get(attr_key)
     if artifact_meta is not None:
         return json.loads(artifact_meta)
 
+    # Search trial_system_attrs. Note that artifacts uploaded via optuna.artifacts.upload_artifact
+    # have a different trial_system_attrs key prefix.
     # See https://github.com/optuna/optuna/blob/f827582a8/optuna/artifacts/_upload.py#L71
     trial_system_attrs = storage.get_trial_system_attrs(trial_id)
-    value = trial_system_attrs.get("artifacts:" + artifact_id)
+    value = trial_system_attrs.get(
+        DASHBOARD_ARTIFACTS_ATTR_PREFIX + artifact_id
```

Comment:
[question] Do we really need to `get(DASHBOARD_ARTIFACTS_ATTR_PREFIX + artifact_id)` because it seems that artifacts are uploaded with `ARTIFACTS_ATTR_PREFIX` from both optuna and optuna-dashboard?

https://github.com/optuna/optuna-dashboard/pull/564/files#diff-b37ec658ac0e92728ec1166294ddd30cf515cbb022e7d88fc970ab99bb8e6f82R113

Reply:
I noticed that it doesn't. Thanks for your comment.